airbyte-integrations/connectors/source-marketo/source_marketo/source.py

```diff
@@ -0,0 +1,530 @@
+        start_date = min(start_date, end_date)
+        date_slices = []
+
+        while start_date <= end_date:
+            end_date_slice = start_date.add(days=self.window_in_days)
+            date_slice = {"startAt": to_datetime_str(start_date), "endAt": to_datetime_str(end_date_slice)}
+
+            date_slices.append(date_slice)
+            start_date = end_date_slice
+
+        return date_slices
+
+
+class MarketoExportBase(IncrementalMarketoStream):
+    """
+    Base class for all the streams which support bulk extract.
+    """
+
+    job_timeout = 60 * 180
+    poll_interval = 60
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    @property
+    def stream_fields(self):
+        return {}
+
+    @property
+    def stream_filter(self):
+        return {}
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"/bulk/v1/{self.stream_name}/export/{stream_slice['id']}/file.json"
+
+    def stream_slices(self, sync_mode, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
+        date_slices = super().stream_slices(sync_mode, stream_state, **kwargs)
+
+        for date_slice in date_slices:
+            param = {"fields": [], "filter": {"createdAt": date_slice}}
+            param["fields"].extend(self.stream_fields)
+            param["filter"].update(self.stream_filter)
+
+            export = next(MarketoExportCreate(self.config, stream_name=self.stream_name, param=param).read_records(sync_mode=None), {})
+
+            date_slice["id"] = export["exportId"]
+        return date_slices
+
+    def sleep_till_export_completed(self, stream_slice: Mapping[str, Any]) -> bool:
+        timeout_time = pendulum.now().add(seconds=self.job_timeout)
+        while pendulum.now() < timeout_time:
+            status = next(
+                MarketoExportStatus(self.config, stream_name=self.stream_name, export_id=stream_slice["id"]).read_records(sync_mode=None)
+            )
+            self.logger.info(f"Export {self.name} from {stream_slice['startAt']} to {stream_slice['endAt']} status is {status}")
+
+            if status == "Created":
+                # If the status is created, the export has been made but
+                # not started, so enqueue the export.
+                next(
+                    MarketoExportStart(self.config, stream_name=self.stream_name, export_id=stream_slice["id"]).read_records(sync_mode=None)
+                )
+
+            elif status in ["Cancelled", "Failed"]:
+                # Cancelled and failed exports fail the current sync.
+                raise Exception(status)
+
+            elif status == "Completed":
+                return True
+
+            sleep(self.poll_interval)
+
+        raise Exception(f"Export timed out after {self.job_timeout / 60} minutes")
```

Comment:
If it's divided by `poll_interval` we can reuse this constant, instead of "60"

Reply:
Updated.